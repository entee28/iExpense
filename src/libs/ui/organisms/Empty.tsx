import { ListEmptyIndicator } from 'assets/icons'
import React, { memo } from 'react'
import { useTranslation } from 'react-i18next'
import { ViewProps, ViewStyle } from 'react-native'
import { Box, Text } from '../atoms'
import colors from '../colors'
import { SCREEN_PADDING_HORIZONTAL } from '../constants'

type WEmptyProps = {
  message?: string
  viewProps?: ViewProps
} & ViewStyle
export const Empty = memo((props: WEmptyProps) => {
  const { message, viewProps, ...style } = props
  const { t } = useTranslation()

  return (
    <Box
      flex={1}
      alignItems="center"
      justifyContent="center"
      paddingBottom={55}
      paddingHorizontal={SCREEN_PADDING_HORIZONTAL}
      viewProps={viewProps}
      {...style}>
      <ListEmptyIndicator />

      <Text
        textAlign="center"
        marginTop={16}
        fontSize={16}
        lineHeight={24}
        color={colors.mono40}>
        {message ?? t('common.empty_list_message')}
      </Text>
    </Box>
  )
})
