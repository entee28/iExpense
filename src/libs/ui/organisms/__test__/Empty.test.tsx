import { render } from '@testing-library/react-native'
import { Empty } from '../Empty'

it('should match snapshot', () => {
  const snapshot = render(<Empty />).toJSON()
  expect(snapshot).toMatchSnapshot()
})
