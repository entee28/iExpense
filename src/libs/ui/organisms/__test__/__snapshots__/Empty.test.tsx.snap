// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 55,
      "paddingHorizontal": 24,
    }
  }
>
  <SvgMock />
  <Text
    style={
      [
        {
          "color": "#1A1A1A66",
          "fontFamily": "Nunito-Regular",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "center",
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    common.empty_list_message
  </Text>
</View>
`;
